tests/components/trafikverket_ferry/conftest.py

```diff
@@ -0,0 +1,80 @@
+"""Fixtures for Trafikverket Ferry integration tests."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from unittest.mock import patch
+
+import pytest
+from pytrafikverket.trafikverket_ferry import FerryStop
+
+from homeassistant.components.trafikverket_ferry.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.core import HomeAssistant
+from homeassistant.util import dt
+
+from . import ENTRY_CONFIG
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture()
+async def load_int(
+    hass: HomeAssistant, get_ferries: list[FerryStop]
+) -> MockConfigEntry:
+    """Set up the Sensibo integration in Home Assistant."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        source=SOURCE_USER,
+        data=ENTRY_CONFIG,
+        entry_id="1",
+        unique_id="123",
+    )
+
+    config_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.trafikverket_ferry.coordinator.TrafikverketFerry.async_get_next_ferry_stops",
+        return_value=get_ferries,
+    ):
```

Comment:
To not risk flaky tests, should this `yield` instead so the patch is in effect for the duration of the test?

Reply:
Sorry but then I don't understand what you mean by this?

Reply:
I meant that `async_get_next_ferry_stops` is only patched during the call to `async_setup`.
If that's the intention, that's perfectly fine and you don't need to change anything,
Otherwise, this fixture should yield the `config_entry` from within the patch's context instead of returning the `config_entry`.

Reply:
Alright. Then we leave it as it's not used for more than setting up the config entry for test_sensor.